core/include/detray/tools/base_stepper.hpp

```diff
@@ -68,8 +70,41 @@ class base_stepper {
 
         navigation_direction _nav_dir = e_forward;
 
+        /// Remaining path length
+        scalar _path_limit = std::numeric_limits<scalar>::max();
+
+        /// Current step size
+        scalar _step_size = std::numeric_limits<scalar>::infinity();
+
+        /// @returns this states remaining path length.
+        DETRAY_HOST_DEVICE
+        inline scalar dist_to_path_limit() const { return _path_limit; }
+
+        /// Set the path limit to a scalar @param pl
+        DETRAY_HOST_DEVICE
+        inline void set_path_limit(const scalar pl) { _path_limit = pl; }
+
+        /// Update and check the path limit against a new @param step size.
+        DETRAY_HOST_DEVICE
+        inline bool check_path_limit() {
+            _path_limit -= _step_size;
+            if (_path_limit <= 0.) {
+                return false;
+            }
+            return true;
+        }
+
+        /// @returns the current step size of this state.
+        DETRAY_HOST_DEVICE
+        inline scalar step_size() const { return _step_size; }
+
+        /// Set next step size
+        DETRAY_HOST_DEVICE
+        inline void set_step_size(const scalar step) { _step_size = step; }
+
+        /// Set next step size and release constrained stepping
         DETRAY_HOST_DEVICE
-        void release_step_size(const scalar /*step*/) {}
+        void release_step_size() {}
```

Comment:
Do we need it anymore?

Reply:
Its there to implement step constraints in the future, but i can remove it for now